File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/epg/retrofit/EpgRequestBodyConverter.java
package org.dvbviewer.controller.data.epg.retrofit;

import org.dvbviewer.controller.entities.EpgEntry;

import java.util.List;

import okhttp3.MediaType;
import okhttp3.RequestBody;
import retrofit2.Converter;

final class EpgRequestBodyConverter implements Converter<List<EpgEntry>, RequestBody> {

    private static final MediaType MEDIA_TYPE = MediaType.parse("text");

    @Override public RequestBody convert(List<EpgEntry> value) {
        return RequestBody.create(MEDIA_TYPE, value.toString());
    }

}

File: dvbViewerController/src/main/java/org/dvbviewer/controller/data/epg/retrofit/EpgRequestBodyConverter.kt
package org.dvbviewer.controller.data.epg.retrofit

import okhttp3.MediaType
import okhttp3.RequestBody
import org.dvbviewer.controller.entities.EpgEntry
import retrofit2.Converter

internal class EpgRequestBodyConverter : Converter<List<EpgEntry>, RequestBody> {

    override fun convert(value: List<EpgEntry>): RequestBody {
        return RequestBody.create(MEDIA_TYPE, value.toString())
    }

    companion object {

        private val MEDIA_TYPE = MediaType.parse("text")
    }

}